src/transport/SecureSession.h

```diff
@@ -231,26 +263,50 @@ class SecureSession : public Session, public ReferenceCounted<SecureSession, Sec
 private:
     enum class State : uint8_t
     {
+        //
+        // Denotes a secure session object that is internally
+        // reserved by the stack before and during session establishment.
+        //
         // Although the stack can tolerate eviction of these (releasing one
         // out from under the holder would exhibit as CHIP_ERROR_INCORRECT_STATE
         // during CASE or PASE), intent is that we should not and would leave
         // these untouched until CASE or PASE complete.
         //
-        // During this stage, the reference counter is hold by the PairingSession
-        kPairing = 1,
+        // In this state, the reference count is held by the PairingSession.
+        //
+        kEstablishing = 1,
 
-        // The session is active, ready for use. During this stage, the
-        // reference counter increased by 1 in Activate, and will be decreased
-        // by 1 when MarkForRemoval is called.
+        //
+        // The session is active, ready for use. When transitioning to this state, the
+        // reference count is incremented by 1 in Activate, and will subsequently be decremented
+        // by 1 when MarkForEviction is called. This ensures the session remains resident
+        // and active for future use even if there currently are no references to it.
+        //
         kActive = 2,
 
-        // The session is pending for removal, all SessionHolders are already
-        // cleared during MarkForRemoval, no future SessionHolder is able to
-        // grab this session, when all SessionHandles go out of scope, the
-        // session object will be released automatically.
-        kPendingRemoval = 3,
+        //
+        // The session is temporarily disabled due to suspicion of a loss of synchronization
+        // with the session state on the peer (e.g transport failure).
+        // In this state, no new exchanges can be created. However, if we receive valid messages
```

Comment:
I see nothing preventing creation of a new _local_ exchange for a defunct session...  Shouldn't there be something?

Reply:
Doesn't this suffice?

```
ExchangeContext * ExchangeManager::NewContext(const SessionHandle & session, ExchangeDelegate * delegate)
{
    if (!session->IsActiveSession())
    {
        // Disallow creating exchange on an inactive session
        ChipLogError(ExchangeManager, "NewContext failed: session inactive");
        return nullptr;
    ...
```

Reply:
Ah, I did not realize that was already there.  Yes, that works.